components/MultiFileDownloader.tsx

```diff
@@ -178,7 +180,23 @@ export async function* traverseFolder(path: string): AsyncGenerator<
     const itemLists = await Promise.all(
       folderPaths.map(fp =>
         (async fp => {
-          const data = await fetcher(`/api?path=${fp}`, hashedToken ?? undefined)
+          let data: any
+          try {
+            data = await fetcher(`/api?path=${fp}`, hashedToken ?? undefined)
+          } catch (error: any) {
+            console.log(error)
+            // 4xx errors are identified as handleable errors
+            if (Math.floor(error.status / 100) === 4) {
+              return {
+                path: fp,
+                isFolder: true,
+                error: [error.status as number, error.message.error as string]
```

Comment:
pass this `error` as an Object instead of an array? or is there a specific reason that you are using an array here?